src/finder.lua

```diff
@@ -302,7 +311,7 @@ function finder.findLegendaryRoot()
 
     -- As of the time of writing this, Legendary is only supported for Windows and Linux.
     -- It follows XDG_CONFIG_HOME and ~/.config/legendary on all platforms.
-    local root = fs.joinpath(os.getenv("XDG_CONFIG_HOME") or fs.joinpath(userOS == "Windows" and sharp.getEnv("USERPROFILE"):result() or os.getenv("HOME"), ".config"), "legendary")
+    local root = fs.joinpath(getLinuxConfigDir() or fs.joinpath(userOS == "Windows" and sharp.getEnv("USERPROFILE"):result(), ".config"), "legendary")
```

Comment:
I don't think you can reach the part after the "or" here, since getLinuxConfigDir() seems to always return something

Reply:
The solution might be to place the windows check first, it would look like this :
`root = fs.joinpath(fs.joinpath(userOS == "Windows" and sharp.getEnv("USERPROFILE"):result(), ".config") or getLinuxConfigDir(), "legendary")`